honeybadgermpc/mpc.py

```diff
@@ -92,47 +108,65 @@ def call_mixin(self, name, *args, **kwargs):
                 value_to_share = self.field.random()
 
             # 'S' is for single shares
-            self.send(j, ('S', shareid, value_to_share))
+            self.send(dest, ('S', shareid, value_to_share))
 
         # Set up the buffer of received shares
         share_buffer = [self._share_buffers[i][shareid] for i in range(self.N)]
 
         point = EvalPoint(self.field, self.N, use_fft=False)
 
         # Create polynomial that reconstructs the shared value by evaluating at 0
-        opening = robust_reconstruct(
+        p, _ = await robust_reconstruct(
```

Comment:
what's the second return value? Why are we ignoring it? Should we log it?

Reply:
The second return value is actually `_confirmed_errors` from reconstruction when successful. I don't think this is worth logging here, but this makes it clearer in the code that the function is actually returning two things. 

That said, there is some error handling to be had here-- this returns `None` in the case of failed reconstruction. We should throw an exception here.

Reply:
Do we add a log when there is an error? If not, then can you please add a log?